### PR TITLE
td: retry on netsplits and server errors

### PR DESCRIPTION
**Pull request by danielnorberg**

### CODE REVIEW

File: digdag-standards/src/main/java/io/digdag/standards/operator/td/TDOperator.java [hunk 1]
```diff
@@ -31,6 +34,12 @@
 public class TDOperator
         implements Closeable
 {
+    private static final Logger logger = LoggerFactory.getLogger(TDOperator.class);
+
+    private static final int INITIAL_RETRY_WAIT = 100;
```
Comment (komamitsu) [inline digdag-standards/src/main/java/io/digdag/standards/operator/td/TDOperator.java]: This initial retry interval seems a bit too short to me. How about 500?

File: digdag-standards/src/main/java/io/digdag/standards/operator/td/TDOperator.java [hunk 2]
```diff
@@ -252,42 +262,99 @@ public TDJobOperator runJob(Config state, String key, JobStarter starter)
...
+            catch (TDClientException e) {
+                logger.warn("failed to start job: domainKey={}", domainKey.get(), e);
+                if (isDeterministicClientException(e)) {
+                    throw e;
+                }
+                throw errorPollingException(state, key, jobState);
+            }
+
+            // Reset error polling
+            jobState = jobState.withErrorPollIteration(Optional.absent());
```
Comment (komamitsu) [inline digdag-standards/src/main/java/io/digdag/standards/operator/td/TDOperator.java]: If you add `jobState.resetErrorPollIteration()` and call it here, it's easier to understand for me.

File: digdag-standards/src/main/java/io/digdag/standards/operator/td/TDOperator.java (hunk 1 above)
Comment (danielnorberg) [inline digdag-standards/src/main/java/io/digdag/standards/operator/td/TDOperator.java]: This is only for the initial ephemeral netsplit handling.

File: digdag-standards/src/main/java/io/digdag/standards/operator/td/TDOperator.java (hunk 2 above)
Comment (danielnorberg) [inline digdag-standards/src/main/java/io/digdag/standards/operator/td/TDOperator.java]: jobState is immutable

Comment (danielnorberg) [inline digdag-standards/src/main/java/io/digdag/standards/operator/td/TDOperator.java]: I guess we could use a mutable builder instead, but I'm not sure I'd prefer that...

File: digdag-standards/src/main/java/io/digdag/standards/operator/td/TDOperator.java (hunk 1 above)
Comment (komamitsu) [inline digdag-standards/src/main/java/io/digdag/standards/operator/td/TDOperator.java]: [minor] Yeah, but I have a little concern that this short retry wait might make the api server busier when td-api is already busy and unstable.

**Comment by komamitsu** [comment]
LGTM with awesome tests 👍  although I felt the initial retry interval for td-api a bit too short.

File: digdag-standards/src/main/java/io/digdag/standards/operator/td/TDOperator.java (hunk 1 above)
Comment (danielnorberg) [inline digdag-standards/src/main/java/io/digdag/standards/operator/td/TDOperator.java]: Per discussion on slack, will change this to two retries with 300 ms initial wait.

File: digdag-standards/src/main/java/io/digdag/standards/operator/td/TDOperator.java (hunk 2 above)
Comment (danielnorberg) [inline digdag-standards/src/main/java/io/digdag/standards/operator/td/TDOperator.java]: Builder doesn't have getters. There's a new `@Value.Modifiable` annotation we could use to generate a value class with setters, but when jackson serializing an object of that class it includes at least one unwanted field. Will stick with the immutable value class for now.

File: digdag-standards/src/main/java/io/digdag/standards/operator/td/TDOperator.java (hunk 1 above)
Comment (danielnorberg) [inline digdag-standards/src/main/java/io/digdag/standards/operator/td/TDOperator.java]: Actually, as some operators still use the retry executor directly without the persistent retry that this PR implements, I'll revert this to the original 3 retries with 500ms initial wait for now. We can change this later when all operators have been state-machinified.

**Pull request merged**